Include/Types/Color.hpp

```diff
@@ -268,171 +268,97 @@ class ColorBase
-		}
+		return _PREDEFINED_COLOR(0.627, 0.125, 0.941);
 	};
 
 	static constexpr ColorBase<T> VIOLET()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.933 * max ), T( 0.509 * max ), T( 0.933 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.933 ), T( 0.509 ), T( 0.933 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.933, 0.509, 0.933);
 	};
 
 	static constexpr ColorBase<T> GREEN()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 1.0 * max ), T( 0.0 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 1.0 ), T( 0.0 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 1.0, 0.0);
 	};
 
 	static constexpr ColorBase<T> LIME()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.196 * max ), T( 0.803 * max ), T( 0.196 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.196 ), T( 0.803 ), T( 0.196 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.196, 0.803, 0.196);
 	};
 
 	static constexpr ColorBase<T> OLIVE()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.419 * max ), T( 0.556 * max ), T( 0.137 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.419 ), T( 0.556 ), T( 0.137 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.419, 0.556, 0.137);
 	};
 
 	static constexpr ColorBase<T> BLUE()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 0.0 * max ), T( 1.0 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 0.0 ), T( 1.0 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 0.0, 1.0);
 	};
 
 	static constexpr ColorBase<T> NAVY()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 0.0 * max ), T( 0.5 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 0.0 ), T( 0.5 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 0.0, 0.5);
 	};
 
 	static constexpr ColorBase<T> CYAN()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 1.0 * max ), T( 1.0 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 1.0 ), T( 1.0 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 1.0, 1.0);
 	};
 
 	static constexpr ColorBase<T> TEAL()
 	{
-		if constexpr( std::is_integral_v<T> ) {
-			T max = std::numeric_limits<T>::max();
-			return ColorBase<T>{ T( 0.0 * max ), T( 0.5 * max ), T( 0.5 * max ), T( 1.0 * max ) };
-		} else {
-			return ColorBase<T>{ T( 0.0 ), T( 0.5 ), T( 0.5 ), T( 1.0 ) };
-		}
+		return _PREDEFINED_COLOR(0.0, 0.5, 0.5);
 	};
 
 	static constexpr ColorBase<T> YELLOW()
```

Comment:
Would it be possible to change predefined colors from functions to static constexpr values?

`YELLOW(){}` -> `YELLOW = PREDEFINED_COLOR()`

Reply:
I can't get it to work that way. It keeps complaining that ColorBase's constructor suddenly is not constexpr when i try:
```
static constexpr ColorBase<T> WHITE = PREDEFINED_COLOR(1.0, 1.0, 1.0);
```

Reply:
Leaving as is then.